python/cuml/test/plugins/quick_run_plugin.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import typing
+
+import pytest
+import _pytest.python
+
+
+def pytest_addoption(parser):
+
+    group = parser.getgroup('Quick Run Plugin')
+
+    group.addoption("--quick_run",
+                    default=False,
+                    action="store_true",
+                    help=("Selecting this option will reduce the number of "
+                          "tests run by only picking tests if one of the "
+                          "parameters has never been seen before."))
+
+
+# This hook must be run last after all others as some plugins may have skipped
+# some tests
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(config, items):
```

Comment:
maybe a more specific name than modifyitems would help clarity

Reply:
The name `pytest_collection_modifyitems` is required. Its a specific hook in pytest that runs after the tests have been collected, but before they have been run. Changing the name changes the hook.